### PR TITLE
feat(api-client, core): Update interfaces for team icons

### PR DESCRIPTION
**Pull request by ffflorian**
## Pull Request Checklist

- [ ] My code is covered by tests
- [x] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/api-client/src/main/team/team/NewTeamData.ts [hunk 1]
```diff
@@ -23,5 +23,6 @@ export interface NewTeamData {
   members?: MemberData[];
   name: string;
   icon: string;
+  icon_key: string;
```
Comment (Yserz) [inline packages/api-client/src/main/team/team/NewTeamData.ts]: I think it should be optional

File: packages/api-client/src/main/team/team/TeamData.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ export interface TeamData {
   creator: string;
   currency?: SupportedCurrency;
   icon: string;
+  icon_key: string;
```
Comment (Yserz) [inline packages/api-client/src/main/team/team/TeamData.ts]: I think it should be optional

File: packages/api-client/src/main/team/team/NewTeamData.ts [hunk 3]
```diff
@@ -20,8 +20,13 @@
...
 export interface NewTeamData {
+  /** User binding team */
+  binding?: boolean;
+  /** Initial team member IDs (between 1 and 127) */
   members?: MemberData[];
   name: string;
+  /** Team icon (asset ID) */
   icon: string;
-  binding?: boolean;
+  /** Team icon asset key */
```
Comment (bennyn) [inline packages/api-client/src/main/team/team/NewTeamData.ts]:
```suggestion
  /** Team icon (asset key) */
```

File: packages/api-client/src/main/team/team/TeamData.ts [hunk 4]
```diff
@@ -20,10 +20,15 @@
...
 
 export interface TeamData {
+  /** User binding team */
   binding: boolean;
+  /** Team creator's user ID */
   creator: string;
   currency?: SupportedCurrency;
+  /** Team icon (asset ID) */
   icon: string;
+  /** Team icon asset key */
```
Comment (bennyn) [inline packages/api-client/src/main/team/team/TeamData.ts]:
```suggestion
  /** Team icon (asset key) */
```

File: packages/api-client/src/main/team/team/NewTeamData.ts [hunk 5]
```diff
@@ -20,8 +20,13 @@
 import {MemberData} from '../member/MemberData';
 
 export interface NewTeamData {
+  /** User binding team */
+  binding?: boolean;
+  /** Initial team member IDs (between 1 and 127) */
```
Comment (bennyn) [inline packages/api-client/src/main/team/team/NewTeamData.ts]: Are the IDs between 1-127 or are there 1-127 batched team member data sets?

File: packages/api-client/src/main/team/team/NewTeamData.ts [hunk 6]
```diff
@@ -20,8 +20,13 @@
 import {MemberData} from '../member/MemberData';
 
 export interface NewTeamData {
+  /** User binding team */
+  binding?: boolean;
```
Comment (bennyn) [inline packages/api-client/src/main/team/team/NewTeamData.ts]: What does it mean to be a "User binding team"?

File: packages/api-client/src/main/team/team/NewTeamData.ts (hunk 5 above)
Comment (ffflorian) [reply packages/api-client/src/main/team/team/NewTeamData.ts]: The IDs are between 1 and 127.

File: packages/api-client/src/main/team/team/NewTeamData.ts (hunk 6 above)
Comment (ffflorian) [reply packages/api-client/src/main/team/team/NewTeamData.ts]: If it's not "binding", it's a legacy team.

Comment (Yserz) [reply packages/api-client/src/main/team/team/NewTeamData.ts]: It stems from the initial idea of teams. There you were able to bind an account to multiple teams.

**Pull request merged**